### PR TITLE
Add a Magic Leap back end

### PR DESCRIPTION
**Pull request by asajeffrey**
This PR ports over the magicleap back end from rust-webvr.

I also did a few drive-by fixes, none of which are very exciting.

### CODE REVIEW

File: webxr/magicleap/mod.rs [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+        device.start_frame()?;
+
+        Ok(device)
+    }
+}
+
+impl MagicLeapDevice {
+    fn start_frame(&mut self) -> Result<(), MLResult> {
+        if !self.in_frame {
+            warn!("Starting frame");
```
[the remark this answers is not in the document]
Comment (Manishearth) [reply webxr/magicleap/mod.rs]: (probably should be info everywhere)

File: webxr/magicleap/mod.rs [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+            };
+
+            warn!("Started frame");
+            self.in_frame = true;
+        }
+        Ok(())
+    }
+
+    // Approximate the viewer transform by linear interpolation of the two eyes
+    fn lerp_transforms(&self) -> RigidTransform3D<f32, Viewer, Native> {
```
Comment (Manishearth) [inline webxr/magicleap/mod.rs]: ugh but also i see why we need this (Josh needs it too for openxr)

File: webxr/magicleap/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (asajeffrey) [reply webxr/magicleap/mod.rs]: Oops, that's a leftover from me debugging.

File: webxr/magicleap/mod.rs (hunk 2 above)
Comment (asajeffrey) [reply webxr/magicleap/mod.rs]: Indeed. If we find ourselves doing this often, we may need to rethink the API.

**Pull request merged**